app/app.go

```diff
@@ -253,54 +259,37 @@ func (app *App) initServices(ctx context.Context) error {
 	app.setStatus(statusInitializing)
 	app.logger.Infof("initialize services...")
 
-	wg := &sync.WaitGroup{}
-	errors := make(chan error, len(app.services))
+	var err error
 
-	wg.Add(len(app.services))
 	for _, svc := range app.services {
-		go func(svc interface{}) {
-			defer wg.Done()
-
-			// Register checks before starting initialization
-			if check, ok := svc.(Checkable); ok {
-				if err := check.RegisterCheck(app.readiness); err != nil {
-					errors <- err
-					return
-				}
+		// Register checks before starting initialization
+		if check, ok := svc.(Checkable); ok {
+			if err = check.RegisterCheck(app.readiness); err != nil {
```

Comment:
I am not we should RegisterCheck before `Init(...)` as the check itself likely requires the service to be initialized.

Note that the health server starts serving right after constructing the app, so it is possible that a call to liveness or readiness comes in before initialization has been completed.

Reply:
Something that could be done is the `App` can protect the services checks from being called if `Init`has not yet completed.

Reply:
This is not something i've changed in this PR. The order stay the same as before, so no potential BC.
Let's change method order in another PR then.

Reply:
created ticket ETH-441